core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -4004,6 +4471,99 @@ class KafkaApis(val requestChannel: RequestChannel,
     CompletableFuture.completedFuture[Unit](())
   }
 
+
+  def getAcknowledgeBatchesFromShareFetchRequest(
+                                                  shareFetchRequest : ShareFetchRequest,
+                                                  topicNames : util.Map[Uuid, String],
+                                                  erroneous : mutable.Map[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData],
+                                                ) : mutable.Map[TopicIdPartition, util.List[ShareAcknowledgementBatch]] = {
+
+    val acknowledgeBatchesMap = mutable.Map[TopicIdPartition, util.List[ShareAcknowledgementBatch]]()
+    shareFetchRequest.data().topics().forEach ( topic => {
+
+      if(!topicNames.asScala.contains(topic.topicId)) {
+        topic.partitions.forEach((partition: ShareFetchRequestData.FetchPartition) => {
+          val topicIdPartition = new TopicIdPartition(
+            topic.topicId,
+            new TopicPartition(null, partition.partitionIndex))
+          erroneous +=
+            topicIdPartition -> ShareAcknowledgeResponse.partitionResponse(topicIdPartition, Errors.UNKNOWN_TOPIC_ID)
+        })
+      }
+      else {
+        topic.partitions().forEach ( partition => {
+          val topicIdPartition = new TopicIdPartition(
+            topic.topicId(),
+            new TopicPartition(topicNames.get(topic.topicId()), partition.partitionIndex())
+          )
+          var exceptionThrown = false
+          val acknowledgeBatches = new util.ArrayList[ShareAcknowledgementBatch]()
+          breakable{
+            partition.acknowledgementBatches().forEach( batch => {
+              try {
+                acknowledgeBatches.add(new ShareAcknowledgementBatch(
+                  batch.firstOffset(),
+                  batch.lastOffset(),
+                  batch.acknowledgeTypes()
+                ))
+              } catch {
+                case e : IllegalArgumentException =>
+                  exceptionThrown = true
+                  erroneous += topicIdPartition -> ShareAcknowledgeResponse.partitionResponse(topicIdPartition, Errors.forException(e))
+                  break
+              }
```

Comment:
What exception is being thrown?

Reply:
This was a mistake, will remove the try catch block. Thanks!